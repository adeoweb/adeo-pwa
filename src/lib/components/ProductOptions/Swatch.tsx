import React, { FunctionComponent } from 'react';
import { useSwatch } from '@magento/peregrine/lib/talons/ProductOptions/useSwatch';
import { TConfigurableProductOptionsValues } from 'src/lib/types/graphql/Product';

type TSwatchProps = {
    isSelected: boolean;
    item: TConfigurableProductOptionsValues;
    onClick: (selection: number) => void;
};

const Swatch: FunctionComponent<TSwatchProps> = ({
    isSelected = false,
    item: { label, value_index },
    onClick
}) => {
    const talonProps = useSwatch({
        onClick,
        value_index
    });

    const { handleClick } = talonProps;

    const style = {
        // TODO: temp, representational solution
        // refactor to use actual color code/image from backend
        backgroundColor: (label || '').toLowerCase()
    };

    return (
        <li className={isSelected ? 'active' : ''}>
            <button style={style} onClick={handleClick} title={label} />
        </li>
    );
};

export default Swatch;
