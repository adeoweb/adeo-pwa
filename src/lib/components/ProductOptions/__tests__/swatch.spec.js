import React from 'react';
import testRenderer from 'react-test-renderer';

import Swatch from '../Swatch';

const defaultProps = {
    item: {
        label: 'red',
        value_index: 0
    }
};

test('renders a Swatch correctly', () => {
    const component = testRenderer.create(<Swatch {...defaultProps} />);

    expect(component.root.findByType('li').props.className).not.toContain(
        'active'
    );

    expect(component).toMatchSnapshot();
});

test('appends "active" to className if isSelected is true', () => {
    const props = {
        ...defaultProps,
        isSelected: true
    };

    const component = testRenderer.create(<Swatch {...props} />);

    expect(component.root.findByType('li').props.className).toContain('active');

    expect(component).toMatchSnapshot();
});
