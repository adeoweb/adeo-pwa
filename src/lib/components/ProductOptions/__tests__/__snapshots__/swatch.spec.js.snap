// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appends "active" to className if isSelected is true 1`] = `
<li
  className="active"
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
      }
    }
    title="red"
  />
</li>
`;

exports[`renders a Swatch correctly 1`] = `
<li
  className=""
>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "red",
      }
    }
    title="red"
  />
</li>
`;
